File: keychain_access/tests/hackme.m
#import <Foundation/Foundation.h>
#import <Security/Security.h>

int main(int argc, const char * argv[]) {
    sleep(1);
    NSDictionary *query = @{(id)kSecClass: (id)kSecClassGenericPassword,
                            (id)kSecAttrLabel: @"org.owlink.findme",
                            (id)kSecReturnData: (id)kCFBooleanTrue};
    
    CFTypeRef item;
    OSStatus res = SecItemCopyMatching((CFDictionaryRef)query, &item);
    if (res != 0) {
        NSLog(@"Failed to extract item: %@", SecCopyErrorMessageString(res, NULL));
    }
    return 0;
}

File: keychain_access/tests/hackme.swift
#!/usr/bin/swift

import Foundation
import Security

sleep(1) // so Frida has time to attach

enum KeychainError : Error {
    case Open(message: String)
    case Unlock(message: String)
    case ItemNotFound
}

// Query password from keychain
let query: [String: Any] = [kSecClass as String: kSecClassGenericPassword,
                            kSecAttrLabel as String: "org.owlink.findme",
                            kSecReturnData as String: true]
var item: CFTypeRef?
let status = SecItemCopyMatching(query as CFDictionary, &item)
guard status == errSecSuccess else { throw KeychainError.ItemNotFound }
